irma/cmd/keyshare-server.go

```diff
@@ -0,0 +1,114 @@
+	flags.String("db", "", "Database server connection string")
+	flags.Lookup("db-type").Header = `Database configuration`
+
+	flags.String("jwt-privkey", "", "Private jwt key of keyshare server")
+	flags.String("jwt-privkey-file", "", "Path to file containing private jwt key of keyshare server")
+	flags.Int("jwt-privkey-id", 0, "Key identifier of keyshare server public key matching used private key")
+	flags.String("jwt-issuer", keysharecore.JWTIssuerDefault, "JWT issuer used in \"iss\" field")
+	flags.Int("jwt-pin-expiry", keysharecore.JWTPinExpiryDefault, "Expiry of PIN JWT in seconds")
+	flags.String("storage-primary-keyfile", "", "Primary key used for encrypting and decrypting secure containers")
+	flags.StringSlice("storage-fallback-keyfile", nil, "Fallback key(s) used to decrypt older secure containers")
+	flags.Lookup("jwt-privkey").Header = `Cryptographic keys`
+
+	flags.String("keyshare-attribute", "", "Attribute identifier that contains username")
+	flags.Lookup("keyshare-attribute").Header = `Keyshare server attribute issued during registration`
+
+	flags.String("email-server", "", "Email server to use for sending email address confirmation emails")
+	flags.String("email-hostname", "", "Hostname used in email server tls certificate (leave empty when mail server does not use tls)")
+	flags.String("email-username", "", "Username to use when authenticating with email server")
+	flags.String("email-password", "", "Password to use when authenticating with email server")
+	flags.String("email-from", "", "Email address to use as sender address")
+	flags.String("default-language", "en", "Default language, used as fallback when users prefered language is not available")
+	flags.StringToString("registration-email-subjects", nil, "Translated subject lines for the registration email")
+	flags.StringToString("registration-email-files", nil, "Translated emails for the registration email")
+	flags.StringToString("verification-url", nil, "Base URL for the email verification link (localized)")
+	flags.Lookup("email-server").Header = `Email configuration (leave empty to disable sending emails)`
+
+	flags.String("tls-cert", "", "TLS certificate (chain)")
+	flags.String("tls-cert-file", "", "path to TLS certificate (chain)")
+	flags.String("tls-privkey", "", "TLS private key")
+	flags.String("tls-privkey-file", "", "path to TLS private key")
+	flags.Bool("no-tls", false, "Disable TLS")
+	flags.Lookup("tls-cert").Header = `TLS configuration (leave empty to disable TLS)`
+
+	flags.CountP("verbose", "v", "verbose (repeatable)")
+	flags.BoolP("quiet", "q", false, "quiet")
+	flags.Bool("log-json", false, "Log in JSON format")
+	flags.Bool("production", false, "Production mode")
+	flags.Lookup("verbose").Header = `Other options`
+}
+
+func configureKeyshared(cmd *cobra.Command) *keyshareserver.Configuration {
+	readConfig(cmd, "keyshareserver", "keyshareserver", []string{".", "/etc/keyshareserver"}, nil)
```

Comment:
Isn't it more logical to make an optional configuration option for other base dirs as `.` (like the `/etc/keyshareserver`)? It seems a bit weird to fix those paths in the code.

Reply:
Right, I didn't know the `irmaserver` also did this. Makes sense to remain the consistency for now then indeed.